### PR TITLE
Test LocalizationManager

### PR DESCRIPTION
**Pull request by daullmer**
**Changes**
Test coverage only goes one way 📈
Order LocalizationManager so public methods are before private methods
Use using declarations
Add 1 to index instead of using `TrimStart(':')`

**Issues**

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Xunit tests don't have a strict ordering

**Comment by Ullmie02** [comment]
@Bond-009 how should I order them?

**Comment by Bond-009** [comment]
Imo best solution would be making _localizationManager a local in each test instead of a field

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            var localizationManager = Setup(new ServerConfiguration
+            {
+                UICulture = "de-DE"
+            });
+            var countries = localizationManager.GetCountries();
+            var countryInfos = countries.ToList();
+
+            Assert.Equal(139, countryInfos.Count);
+
+            var germany = countryInfos.FirstOrDefault(x => x.Name == "DE");
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:26]: use `string.Equals` with `StringComparison.Ordinal`

File: Emby.Server.Implementations/Localization/LocalizationManager.cs [hunk 2]
```diff
@@ -51,120 +51,13 @@ public class LocalizationManager : ILocalizationManager
             _logger = logger;
         }
 
-        /// <summary>
-        /// Loads all resources into memory.
-        /// </summary>
-        /// <returns><see cref="Task" />.</returns>
-        public async Task LoadAll()
```
Comment (Bond-009) [inline Emby.Server.Implementations/Localization/LocalizationManager.cs:58]: Why move these around?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/Localization/LocalizationManager.cs (hunk 2 above)
Comment (daullmer) [reply Emby.Server.Implementations/Localization/LocalizationManager.cs:59]: I have no idea what I wanted to do here so I reverted the commit and made some small improvements

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    {
+        [Fact]
+        public void GetCountries_All_Success()
+        {
+            var localizationManager = Setup(new ServerConfiguration
+            {
+                UICulture = "de-DE"
+            });
+            var countries = localizationManager.GetCountries();
+            var countryInfos = countries.ToList();
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:23]: Why 2 steps?

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            var localizationManager = Setup(new ServerConfiguration
+            {
+                UICulture = "de-DE"
+            });
+            await localizationManager.LoadAll();
+            var ratings = localizationManager.GetParentalRatings().ToList();
+
+            Assert.Equal(23, ratings.Count);
+
+            var tvma = ratings.FirstOrDefault(x => x.Name == "TV-MA");
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:89]: Use string.Equals here

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            var localizationManager = Setup(new ServerConfiguration()
+            {
+                MetadataCountryCode = "DE"
+            });
+            await localizationManager.LoadAll();
+            var ratings = localizationManager.GetParentalRatings().ToList();
+
+            Assert.Equal(10, ratings.Count);
+
+            var fsk = ratings.FirstOrDefault(x => x.Name == "FSK-12");
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:106]: Same

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by Bond-009 — APPROVED** [review]

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+            Assert.Equal("German", germany.Name);
+            Assert.Contains("deu", germany.ThreeLetterISOLanguageNames);
+            Assert.Contains("ger", germany.ThreeLetterISOLanguageNames);
+        }
+
+        [Theory]
+        [InlineData("de")]
+        [InlineData("ger")]
+        [InlineData("german")]
+        public async Task FindLanguage_Valid_Success(string identifier)
```
Comment (cvium) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:58]:
```suggestion
        public async Task FindLanguageInfo_Valid_Success(string identifier)
```

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+            Assert.Equal("ger", germany.ThreeLetterISOLanguageName);
+            Assert.Equal("German", germany.DisplayName);
+            Assert.Equal("German", germany.Name);
+            Assert.Contains("deu", germany.ThreeLetterISOLanguageNames);
+            Assert.Contains("ger", germany.ThreeLetterISOLanguageNames);
+        }
+
+        [Fact]
+        public async Task ParentalRatings_Default_Success()
```
Comment (cvium) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:77]:
```suggestion
        public async Task GetParentalRatings_Default_Success()
```

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+            Assert.Equal(23, ratings.Count);
+
+            var tvma = ratings.FirstOrDefault(x => x.Name.Equals("TV-MA", StringComparison.Ordinal));
+            Assert.NotNull(tvma);
+            Assert.Equal(9, tvma!.Value);
+        }
+
+        [Fact]
+        public async Task ParentalRatings_ConfiguredCountryCode_Success()
```
Comment (cvium) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:94]:
```suggestion
        public async Task GetParentalRatings_ConfiguredCountryCode_Success()
```

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+        [Theory]
+        [InlineData("CA-R", "CA", 10)]
+        [InlineData("FSK-16", "DE", 8)]
+        [InlineData("FSK-18", "DE", 9)]
+        [InlineData("FSK-18", "US", 9)]
+        [InlineData("TV-MA", "US", 9)]
+        [InlineData("XXX", "asdf", 100)]
+        [InlineData("Germany: FSK-18", "DE", 9)]
+        public async Task GetRatingLevelFromString_Valid_Success(string value, string countryCode, int expectedLevel)
```
Comment (cvium) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:118]:
```suggestion
        public async Task GetRatingLevel_GivenValidString_Success(string value, string countryCode, int expectedLevel)
```

File: tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+                MetadataCountryCode = countryCode
+            });
+            await localizationManager.LoadAll();
+            var level = localizationManager.GetRatingLevel(value);
+            Assert.NotNull(level);
+            Assert.Equal(expectedLevel, level!);
+        }
+
+        [Fact]
+        public async Task GetRatingLevelFromString_Unrated_Success()
```
Comment (cvium) [inline tests/Jellyfin.Server.Implementations.Tests/Localization/LocalizationManagerTests.cs:131]:
```suggestion
        public async Task GetRatingLevel_GivenUnratedString_Success()
```

**Review by cvium — APPROVED** [review]

**Pull request merged**